atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -256,29 +256,29 @@ int getMaxTriesTotal() {
     private synchronized void refreshPool() {
         blacklist.checkAndUpdate(cassandra.getPools());
 
-        Set<InetSocketAddress> serversToAdd = Sets.newHashSet(config.servers());
-        Set<InetSocketAddress> serversToRemove = ImmutableSet.of();
+        Set<InetSocketAddress> serversToRemove = Sets.newHashSet();
+        Set<InetSocketAddress> serversToAdd = Sets.newHashSet();
+        Set<InetSocketAddress> currentServers = cassandra.refreshTokenRanges();
 
         if (config.autoRefreshNodes()) {
-            serversToAdd.addAll(cassandra.refreshTokenRanges());
+            serversToRemove.addAll(Sets.difference(cassandra.getPools().keySet(), currentServers));
+            serversToAdd.addAll(Sets.difference(currentServers, cassandra.getPools().keySet()));
         }
 
-        serversToAdd = Sets.difference(serversToAdd, cassandra.getPools().keySet());
-
+        //what does this mean?????
```

Comment:
No idea what the comment was trying to convey, the whole logic here seems to just be a convoluted way of keeping the cassandra pool == config.servers, but we should never be removing anything since nothing else is modifying them and the config is fixed